### PR TITLE
Application Template implements Application

### PR DESCRIPTION
**Pull request by gokulavasan**
CDAP-2142 : ApplicationTemplate implements Application.

Application Template can now add datasets (and streams).

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/LocalApplicationTemplateManager.java [hunk 1]
```diff
@@ -97,6 +98,7 @@ public LocalApplicationTemplateManager(CConfiguration configuration, PipelineFac
     pipeline.addLast(new ApplicationTemplateVerificationStage(store, datasetFramework, adapterService));
     pipeline.addLast(new DeployDatasetModulesStage(configuration, namespace,
                                                    datasetFramework, inMemoryDatasetFramework));
+    pipeline.addLast(new CreateDatasetInstancesStage(configuration, datasetFramework, namespace));
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/LocalApplicationTemplateManager.java]: ApplicationTemplateVerificationStage will have to change too.

Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/LocalApplicationTemplateManager.java]: should we add back stream creation as well?  dont have a use case right now, but if you can create datasets it seems weird that you cant create streams

**Comment by albertshau** [comment]
couple comments otherwise lgtm

**Comment by gokulavasan** [comment]
Build : http://builds.cask.co/browse/CDAP-DUT1473-1

**Comment by albertshau** [comment]
:+1:

**Pull request merged**